### PR TITLE
First version

### PR DESCRIPTION
**Pull request by trocolit2**
Please, review my code. :)
@gustavoneves12
@romulogcerqueira 
@dbcesar

### CODE REVIEW

**Comment by trocolit2** [comment]
[Photo Mosaic: Image Stitching](https://trello.com/c/4wGhWyPR/2-photo-mosaic-image-stitching)

File: src/ImageStitching.cpp [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+                float scale_factor,
+                ImageStitchingMethod *imageStitchingMethod){
+
+    scale_factor_ = scale_factor;
+    imageStitchingMethod_ = imageStitchingMethod;
+
+}
+
+ImageStitching::~ImageStitching(){
+  delete imageStitchingMethod_;
```
Comment (gustavoneves12) [inline src/ImageStitching.cpp]: As far as I know, it is not a good practice delete objects created outside of the class.

File: test/ImageStitching_test.cpp [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+    size_t array_size = ARRAY_SIZE(BOAT_ITERATIONS);
+
+    OpenCVImageStitchingStruct conf;
+    conf.type_warper = CYLINDRICAL_WARPER;
+
+    for (size_t i = 0; i < array_size; i++) {
+        std::cout << " ID TEST = " << BOAT_ITERATIONS[i] << std::endl;
+        size_t number_images = BOAT_ITERATIONS[i].width;
+        size_t iteration = BOAT_ITERATIONS[i].height;
+        cv::Mat stitched_image = imageStitchingTest(
```
Comment (gustavoneves12) [inline test/ImageStitching_test.cpp]: if you remove the delete from ImageStitchingMethod class, you can change here to:

```cplusplus
OpenCVImageStitching stitching_method;
cv::Mat stitched_image = imageStitchingTest(..., &stitching_method);

```

**Comment by gustavoneves12** [comment]
@trocolit2, your code looks nice!

Have you tested using our logs?

**Comment by trocolit2** [comment]
thanks @gustavoneves12!! :+1:

**Pull request closed without merging**